README.md

```diff
@@ -69,3 +69,15 @@ $ forge --help
 $ anvil --help
 $ cast --help
 ```
+
+## Validate Upgrade Safety 
+1. Checkout to the deployed implementation commit 
+2. Copy the YieldDistributor to test/upgrades/<version>/YieldDistributor.sol
+3. Checkout to upgrade candidate version (A version that is strictly higher than the version in the previous step)
+4. Update the version in the options object of the script/upgrades/ValidateUpgrade.s.sol script
+5. Run `forge clean && forge build && forge script script/upgrades/ValidateUpgrade.s.sol`
+6. If script is run succesfully , proceed , otherwise address errors produced by the script unti no errors are produced.
+
+## Test Upgrade with Calldata Locally 
+1. Amend the `data` variable `script/upgrades/UpgradeYieldDistributor.s.sol` to match desired data 
+2. run `forge clean && forge build && forge script script/upgrades/UpgradeYieldDistributor.s.sol --rpc-url $RPC_URL  --sender <proxy_admin>` (proxy admin address is configured to be the breadchain multisig at address 0x918dEf5d593F46735f74F9E2B280Fe51AF3A99ad) 
```

Comment:
proxy admin address is configured to be the Breadchain multisig at address `0x918dEf5d593F46735f74F9E2B280Fe51AF3A99ad`

Reply:
addressed with a7bc3fb